### DISCUSSION TITLE
[Announcement] Homework 0-4

### DISCUSSION THREAD

**Discussion by KihongHeo** [prosyslab-classroom/cs424-program-reasoning#312] [Announcements]
Hi. 
Here is another gift for you. This homework is optional. (similar to HW 0-3)

Read the three articles about the future of programming on the course webpage:
- [Wired](https://www.wired.com/story/ai-write-code-like-humans-bugs/)
- [IEEE Spectrum](https://spectrum.ieee.org/ai-code-generation-language-models)
- [CACM](https://cacm.acm.org/magazines/2022/10/264844-neurosymbolic-ai/fulltext)

and think about the followings:
- What is the future of programming?
- What are the limitations of current AI-based programming?
- How can we overcome the above challenges?

Provide your opinion and write an essay. The same (syntactic and semantic) evaluation criteria will be applied as before.

Due: Nov 11 23:59:59 (no late submission allowed)

TAs, please add the invitation link below.

**Comment by Hhro** [comment]
Here is the invitation link for homework 0-4: https://classroom.github.com/a/TgcDAMTx

You should submit your critique at **Gradescope** in pdf format.

**Comment by KihongHeo** [comment]
For Korean native students, please read [this document](https://github.com/prosyslab/pl-wiki/wiki/번역-용례).
I summarized several awkward translations from your previous submission. You are encouraged to actively use the terms.
